### PR TITLE
Fix room alias lookup vs peeking race condition

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/12307

If the room alias lookup was faster than the initial sync then we would never call `_setupRoom` with `shouldPeek=true` due to a regression caused by https://github.com/matrix-org/matrix-react-sdk/commit/b96ad4a4f09b6edfebf279a6d4ebda65dd04ebdd

Additionally shows the peeking indicator eagerly to set expectations better.

The simpler solution of just re-triggering _setupRoom when shouldPeek stabilized would have worked for this fix too but then would not have worked for showing the peeking loading indicator between the alias loading and the client stabilizing.

### CODE REVIEW

File: src/components/structures/RoomView.js [hunk 1]
```diff
@@ -1663,14 +1676,16 @@ export default createReactClass({
         const ErrorBoundary = sdk.getComponent("elements.ErrorBoundary");
 
         if (!this.state.room) {
-            const loading = this.state.roomLoading || this.state.peekLoading;
+            const loading = this.state.matrixClientIsReady || this.state.roomLoading || this.state.peekLoading;
```
Comment (turt2live) [inline src/components/structures/RoomView.js:1679]: Presumably:
```suggestion
            const loading = !this.state.matrixClientIsReady || this.state.roomLoading || this.state.peekLoading;
```

File: src/components/structures/RoomView.js [hunk 2]
```diff
@@ -1663,14 +1676,16 @@ export default createReactClass({
...
             if (loading) {
+                // Assume preview loading if we don't have a ready client or a room ID (still resolving the alias)
+                const previewLoading = !this.state.matrixClientIsReady || !this.state.roomId || this.state.peekLoading;
```
Comment (turt2live) [inline src/components/structures/RoomView.js:1682]: What happened to the `roomLoading` flag?

File: src/stores/RoomViewStore.js [hunk 3]
```diff
@@ -224,6 +215,7 @@ class RoomViewStore extends Store {
                     storeRoomAliasInCache(payload.room_alias, result.room_id);
                     roomId = result.room_id;
                 } catch (err) {
+                    console.error(err);
```
Comment (turt2live) [inline src/stores/RoomViewStore.js:218]: debug line that can be removed?

Comment (t3chguy) [reply src/stores/RoomViewStore.js:218]: I was intending it as a way to actually put the error into the rageshakes, will add context.

File: src/components/structures/RoomView.js (hunk 2 above)
Comment (t3chguy) [reply src/components/structures/RoomView.js:1682]: roomLoading was never part of `previewLoading`: 
https://github.com/matrix-org/matrix-react-sdk/pull/4606/files#diff-36759beafdbb5cc10e5d7e4ff1d98d87L1673

**Pull request merged**